gapis/api/vulkan/looping_vulkan_control_flow_generator.go

```diff
@@ -98,18 +93,31 @@ type backupMemory struct {
 	offset VkDeviceSize
 }
 
-// Transfrom
-type frameLoop struct {
-	capture   *capture.GraphicsCapture
-	loopCount int32
+type LoopCallbackFunc func(ctx context.Context, request *gapir.FenceReadyRequest)
+
+type LoopCallbacks struct {
+	preLoop  LoopCallbackFunc
+	postLoop LoopCallbackFunc
+}
+
+type loopingVulkanControlFlowGenerator struct {
+	ctx context.Context
```

Comment:
we should not be storing the context in structs. This makes them go from the stack to the heap and is expensive. Any function that reads this value, should just receive a context as a parameter.